tutorials/tutorial-howto-simulate-iot-sensor-data.md

```diff
@@ -0,0 +1,252 @@
+First, connect to your database via `psql`.
+
+Second, create the “sensors” and “sensor_data” tables:
+
+```sql
+CREATE TABLE sensors(
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(50),
+  location VARCHAR(50)
+);
+```
+
+```sql
+CREATE TABLE sensor_data (
+  time TIMESTAMPTZ NOT NULL,
+  sensor_id INTEGER,
+  temperature DOUBLE PRECISION,
+  cpu DOUBLE PRECISION,
+  FOREIGN KEY (sensor_id) REFERENCES sensors (id)
+);
+```
+
+Third (only if you are using TimescaleDB), convert the `sensor_data`
+table into a [`hypertable`][docs-hypertable]:
+
+```sql
+SELECT create_hypertable('sensor_data', 'time');
+```
+
+Fourth, populate the sensors table with 4 sensors:
+
+```sql
+INSERT INTO sensors (type, location) VALUES
+('a','floor'),
+('a', 'ceiling'),
+('b','floor'),
+('b', 'ceiling');
+```
+
+Fifth, verify that the sensors were created correctly:
+
+```sql
+SELECT * FROM sensors;
+```
+
+After running that last SQL statement, you should see something like
+this:
+
+```
+ id | type | location 
+----+------+----------
+  1 | a    | floor
+  2 | a    | ceiling
+  3 | b    | floor
+  4 | b    | ceiling
+(4 rows)
+```
+
+### Step 2 - Create the simulated IoT sensor data [](step2)
+
+*Note: for the following sections we’ll share the results of our queries
+as an example, but since the tutorial generates random data every time
+it is run, your results will look different (but will be structured the
+same way).*
+
+First, generate a dataset for all of our four sensors and insert into
+our sensor_data table:
+
+```sql
+INSERT INTO sensor_data (time, sensor_id, cpu, temperature)
+SELECT
+  time,
+  sensor_id,
+  random() AS cpu,
+  random()*100 AS temperature
+FROM generate_series(now() - interval '24 hour', now(), interval '5 minute') AS g1(time), generate_series(1,4,1) AS g2(sensor_id);
+```
+
+Second, verify that the simulated sensor data was written correctly:
+
+```sql
+SELECT * FROM sensor_data ORDER BY time;
+```
+
+SAMPLE OUTPUT:
+
+```
+             time              | sensor_id |    temperature     |         cpu         
+-------------------------------+-----------+--------------------+---------------------
+ 2020-03-31 15:56:25.843575+00 |         1 |   6.86688972637057 |   0.682070567272604
+ 2020-03-31 15:56:40.244287+00 |         2 |    26.589260622859 |   0.229583469685167
+ 2030-03-31 15:56:45.653115+00 |         3 |   79.9925176426768 |   0.457779890391976
+ 2020-03-31 15:56:53.560205+00 |         4 |   24.3201029952615 |   0.641885648947209
+ 2020-03-31 16:01:25.843575+00 |         1 |   33.3203678019345 |  0.0159163917414844
+ 2020-03-31 16:01:40.244287+00 |         2 |   31.2673618085682 |   0.701185956597328
+ 2020-03-31 16:01:45.653115+00 |         3 |   85.2960689924657 |   0.693413889966905
+ 2020-03-31 16:01:53.560205+00 |         4 |   79.4769988860935 |   0.360561791341752
+...
+```
+
+Congrats! We've created a basic IoT sensor dataset. Now let's run some 
+queries.
+
+### Step 3 - Run basic queries (optional) [](#step3)
```

Comment:
```suggestion
### Step 3 - Run basic queries (optional) [](step3)
```

Reply:
Sorry, I didn't catch these on the first review.  This breaks the TOC links

Reply:
Fixed